crates/wash-cli/src/wit.rs

```diff
@@ -0,0 +1,104 @@
+// This is pretty much a copy of the wkg wit subcommand adapted for wash
+use std::path::PathBuf;
+
+use anyhow::Ok;
+use clap::{Args, Subcommand};
+use wash_lib::{
+    build::monkey_patch_fetch_logging,
+    cli::{CommandOutput, CommonPackageArgs},
+};
+use wasm_pkg_core::{
+    lock::LockFile,
+    wit::{self, OutputType},
+};
+
+/// Commands for interacting with wit
+#[derive(Debug, Subcommand, Clone)]
+pub enum WitCommand {
+    /// Build a WIT package from a directory. By default, this will fetch all dependencies needed
+    /// and encode them in the WIT package. This will generate a lock file that can be used to fetch
+    /// the dependencies in the future.
+    Build(BuildArgs),
+    /// Fetch dependencies for a component. This will read the package containing the world(s) you
+    /// have defined in the given wit directory (`wit` by default). It will then fetch the
+    /// dependencies and write them to the `deps` directory along with a lock file. If no lock file
+    /// exists, it will fetch all dependencies. If a lock file exists, it will fetch any
+    /// dependencies that are not in the lock file and update the lock file.
+    Fetch(FetchArgs),
+}
+
+impl WitCommand {
+    pub async fn run(self) -> anyhow::Result<CommandOutput> {
+        match self {
+            WitCommand::Build(args) => args.run().await,
+            WitCommand::Fetch(args) => args.run().await,
+        }
+    }
+}
+
+#[derive(Debug, Args, Clone)]
+pub struct BuildArgs {
+    /// The directory containing the WIT files to build.
+    #[clap(short = 'd', long = "wit-dir", default_value = "wit")]
+    pub dir: PathBuf,
+
+    /// The name of the file that should be written. This can also be a full path. Defaults to the
+    /// current directory with the name of the package
+    #[clap(short = 'f', long = "file")]
+    pub output_file: Option<PathBuf>,
```

Comment:
nit: Other places in wash we've called this a destination / output. I think this is fine, but we should make them consistent at least in a followup. (this is a good first issue)

Reply:
I was trying to mostly match `wkg wit` here. But I did have to rename the arg anyway because we already have an "output"